lib/utils/wechat-mp.js

```diff
@@ -49,14 +55,89 @@ const detectOriginalArticleUrl = ($) => {
     return null;
 };
 
-const detectSourceUrl = ($) => {
-    const matchs = $.root()
-        .html()
-        .match(/msg_source_url = '(.+)';/);
+/**
+ * Detect source url from the html text of the article.
+ *
+ * @param {string} html - The html text of the article.
+ * @return {(string|null)} - A source url.
+ */
+const detectSourceUrl = (html = '') => {
+    const matchs = html.match(/msg_source_url = '(.+)';/);
+
+    if (matchs) {
+        return matchs[1];
+    }
+    return null;
+};
+
+/**
+ * Detect Voice info from the html text of the article.
+ *
+ * @param {string} html - The html text of the article.
+ * @return {(object|null)} - An object containing the voice id, duration and name.
+ */
+const detectVoiceInfo = (html = '') => {
+    // example: https://mp.weixin.qq.com/s/FY6yQC_e4NMAxK0FBr6jwQ
+    const voiceIdMatchs = html.match(/"voice_id":"([^"]+)",/);
+    const durationMatchs = html.match(/\bduration : "([^"]+)"\*1,/);
+    const nameMatchs = html.match(/\bwindow\.title = "(.+)";/);
+
+    if (voiceIdMatchs) {
+        return {
+            voiceId: voiceIdMatchs[1],
+            duration: durationMatchs ? durationMatchs[1] : '',
+            name: nameMatchs ? nameMatchs[1] : '',
+        };
+    }
+    return null;
+};
+
+/**
+ * Render <audio> html.
+ *
+ * @param {object} options - The info of voice.
+ * @param {string} options.voiceId - The voice id of voice.
+ * @param {string} [options.name] - The name of voice.
+ * @param {string} [options.duration] - The duration of voice .
+ * @return {string} - The <audio> html, a string.
+ */
+const renderAudio = (options = {}) => {
+    if (options.voiceId) {
+        return `<audio controls="controls" preload="auto" duration="${options.duration || ''}"  style="width:100%" src="https://res.wx.qq.com/voice/getvoice?mediaid=${options.voiceId}" title="${options.name || ''}"></audio>`;
+    }
+    return '';
+};
 
+/**
+ * Get head image of the MP account.
+ *
+ * @param {string} html - The html text of the article.
+ * @return {(string|null)} - A head image url.
+ */
+const getAccountHeadImage = (html = '') => {
+    const matchs = html.match(/hd_head_img = "([^"]+)"/);
     if (matchs) {
         return matchs[1];
     }
+
+    const matchs2 = html.match(/getXmlValue\('hd_head_img.DATA'\) \|\| '' : '([^']+)'/);
```

Comment:
```suggestion
    const matchs2 = html.match(/getXmlValue\('hd_head_img.DATA'\) *\|\| *'' *: *'([^']+)'/);
```
To make it more adaptive.